### PR TITLE
executor: support Chunk for StreamAggExec

### PR DESCRIPTION
**Pull request by zz-jason**
to #5261

### CODE REVIEW

**Comment by zz-jason** [comment]
@coocood @XuHuaiyu @spongedu PTAL

File: executor/aggregate.go [hunk 1]
```diff
@@ -226,8 +231,85 @@ func (e *StreamAggExec) Next(goCtx goctx.Context) (Row, error) {
...
+			for i, af := range e.AggFuncs {
+				retRow = append(retRow, af.GetResult(e.aggCtxs[i]))
+				e.aggCtxs[i] = af.CreateContext()
+			}
+			mutableRow.SetDatums(retRow...)
+			if chk.NumCols() == 0 {
+				chk.SetNumVirtualRows(chk.NumRows() + 1)
+			} else {
+				chk.AppendRow(0, mutableRow.ToRow())
+			}
```
Comment (zimulala) [inline executor/aggregate.go]: Could we extract line294-304 into a function? It seems duplicate with line 254-264.

**Comment by coocood** [comment]
LGTM

File: executor/builder.go [hunk 2]
```diff
@@ -672,12 +672,14 @@ func (b *executorBuilder) buildStreamAgg(v *plan.PhysicalStreamAgg) Executor {
...
 		return nil
 	}
-	return &StreamAggExec{
+	e := &StreamAggExec{
 		baseExecutor: newBaseExecutor(v.Schema(), b.ctx, src),
 		StmtCtx:      b.ctx.GetSessionVars().StmtCtx,
 		AggFuncs:     v.AggFuncs,
 		GroupByItems: v.GroupByItems,
 	}
+	e.supportChk = true
```
Comment (ngaut) [inline executor/builder.go]: Move assignment to definition.

Comment (coocood) [reply executor/builder.go]: This is a field in embedded `baseExecutor`, it's  hard to init in the definition.

File: executor/aggregate.go [hunk 3]
```diff
@@ -226,8 +239,92 @@ func (e *StreamAggExec) Next(goCtx goctx.Context) (Row, error) {
...
+// result chunk, and realloc the evaluation context for each aggregation.
+func (e *StreamAggExec) appendResult2Chunk(chk *chunk.Chunk) {
+	e.rowBuffer = e.rowBuffer[:0]
+	for i, af := range e.AggFuncs {
+		e.rowBuffer = append(e.rowBuffer, af.GetResult(e.aggCtxs[i]))
+		e.aggCtxs[i] = af.CreateContext()
+	}
+	e.mutableRow.SetDatums(e.rowBuffer...)
+	if chk.NumCols() == 0 {
+		chk.SetNumVirtualRows(chk.NumRows() + 1)
```
Comment (zimulala) [inline executor/aggregate.go]: Can we add some comments here?

**Comment by zimulala** [comment]
Rest LGTM

File: checkout-pr-branch.sh [hunk 4]
```diff
@@ -16,6 +16,7 @@ fork="https://github.com/$username/tidb"
 
 exists=`git show-ref refs/heads/$local_branch`
 if [ -n "$exists" ]; then
+    git checkout $local_branch
```
Comment (coocood) [inline checkout-pr-branch.sh]: This is an irrelevant change.

**Pull request merged**